specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/components_API.json

```diff
@@ -434,6 +506,63 @@
           "description": "The URI to get the next set of Application Insights component defintions if too many components where returned in the result set."
         }
       }
+    },
+    "ComponentPurgeBody": {
+      "description": "Describes the body of a purge request for an App Insights component",
+      "required": [
+        "table",
+        "filters"
+      ],
+      "properties": {
+        "table": {
+          "type": "string",
+          "description": "Table from which to purge data."
+        },
+        "filters": {
+          "type": "array",
+          "description": "The set of columns and filters (queries) to run over them to purge the resulting data.",
+          "items": {
+            "$ref": "#/definitions/ComponentPurgeBodyFilters"
+          }
+        }
+      }
+    },
+    "ComponentPurgeBodyFilters": {
+      "description": "User-defined filters to return data which will be purged from the table.",
+      "properties": {
+        "column": {
+          "description": "The column of the table over which the given query should run",
+          "type": "string"
+        },
+        "filter": {
+          "description": "A query to to run over the provided table and column to purge the corresponding data.",
+          "type": "string"
+        }
+      }
+    },
+    "ComponentPurgeResponse": {
+      "description": "Response containing operationId for a specific purge action.",
+      "properties": {
+        "operationId": {
+          "description": "Id to use when querying for status for a particular purge operation.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "operationId"
+      ]
+    },
+    "ComponentPurgeStatusResponse": {
+      "description": "Response containing status for a specific purge operation.",
+      "properties": {
+        "status": {
+          "description": "Status of the operation represented by the requested Id.",
+          "type": "string"
```

Comment:
Can this be an enum so that callers are aware of the possible values?

Reply:
Yes it can! 👍